robo_soccer_platform/lib/robo_soccer_platform/game_controller.ex

```diff
@@ -303,50 +336,37 @@ defmodule RoboSoccerPlatform.GameController do
     cooperation_metric_function_name = opts[:cooperation_metric_function_name]
     speed_coefficient = opts[:speed_coefficient]
 
-    robot_connections =
+    team_states =
       opts
       |> Keyword.fetch!(:robot_configs)
-      |> Map.new(fn {team, %{robot_ip_address: ip_address, local_port: local_port}} ->
-        case RobotConnection.start(team, local_port, ip_address) do
-          {:ok, pid} ->
-            Logger.info("Started RobotConnection for team #{team} with pid #{inspect(pid)}")
-            {team, pid}
-
-          {:error, reason} ->
-            raise "Failed starting RobotConnection for team #{team}, reason: #{reason}"
-        end
-      end)
+      |> Map.new(fn {team, {robot_connection_module, robot_connection_config}} ->
+        {:ok, pid} = robot_connection_module.start(team, robot_connection_config)
```

Comment:
Why case was removed from here? Can't this call error?

Reply:
It can, but in practice it won't, plus when we matched to an error we just raised, so there's really no reason for a custom error, MatchError is fine